library/std/src/macros.rs

```diff
@@ -379,3 +379,51 @@ macro_rules! dbg {
         ($($crate::dbg!($val)),+,)
     };
 }
+
+/// Creates a [`HashMap`] containing the arguments.
+///
+/// This macro creates or either an empty [`HashMap`] using
+/// [`HashMap::new`] if there are no arguments, or generates
+/// a [`HashMap::insert`] call for each element.
+///
+/// This macro allows trailing commas.
+///
+/// This macro is used by concatenating comma separated sequences
+/// of `<key:expr> => <value:expr>`.
+///
+/// An example usage of this macro could be the following
+///
+/// ```rust
+/// #![feature(hash_map_macro)]
+///
+/// let map = hash_map!{
+///     "key" => "value",
+///     "key1" => "value1"
+/// };
+///
+/// for (key, value) in map {
+///     println!("{key:?} => {value:?}");
+/// }
```

Comment:
can you replace this with assertions instead? to make it clearer what the result is like (though it is pretty obvious) and also serve as a test.